### ISSUE TITLE
Peer Review (Lab 2)

### ISSUE THREAD

**Issue by FarisanFekri** [LeoDardanello/computational_intelligence#3]
- This code implements a rule as the professor suggested in the customization of the move function. I personally think this an advanced choice as the professor's choose to have one sigma per each individual. It seems to update the performance of the corresponding solution. If you consider the population as a list of percentage of the probabilities then they are not independent because no matter what is the percentage of the first element is, the other elements are constraint to have the rest of the percentage at most, so it was a good choice to disjoint them. 

- Therefore; when it is mutating to create a new offspring, all the probabilities are completely free and they are not bounded so is the sigma. Then it chooses a random number between the length of the first individual which is the number of move the player can have. In the end, it has one probability for each move which is remarkable.

- It was considered to have the move operation on different rows. Each move calculates the row to do the move and in the end it returns an index object which is the number of objects to remove - the peer was careful that the player shouldn't make 0 moves which means he is not playing at all and it would be a disadvantage to the other players.

- This code has the formula from the theory (exponential of the Scaling Factor and a random Gaussian). It picks form a standard Gaussian and selects n rules per each sample and scale each sample with this factor and then raise it to the exponent to avoid having negative values in the variance. The peer mentions that he tried to use an absolute value but he was obtaining worse results using the exponent. It would've been great to find an alternative way to use the absolute value and make it work more efficiently but the main solution is good as well.

- For the next move, there is set a debugging move to check if the ES algorithm is working optimally. If at the end of the generation our move has a higher probability corresponding to the second move, it means it is working and it should be chosen most of the times. It is overall a decent way to check if we are winning or losing the game.

- In summary, the peer has demonstrated both theoretical understanding and practical implementation skills in evolutionary strategies. The code exhibits a balance between complexity and clarity, making it a commendable solution in optimizing strategies for the given problem.
